homeassistant/components/gpslogger/device_tracker.py

```diff
@@ -137,40 +148,53 @@ async def async_added_to_hass(self) -> None:
             self._location = (None, None)
             self._accuracy = None
             self._attributes = {
-                ATTR_ALTITUDE: None,
                 ATTR_ACTIVITY: None,
+                ATTR_ALTITUDE: None,
+                ATTR_BATTERY_CHARGING: None,
                 ATTR_DIRECTION: None,
+                ATTR_LAST_SEEN: None,
                 ATTR_PROVIDER: None,
                 ATTR_SPEED: None,
             }
             self._battery = None
+            self._prv_seen = None
             return
 
         attr = state.attributes
         self._location = (attr.get(ATTR_LATITUDE), attr.get(ATTR_LONGITUDE))
         self._accuracy = attr.get(ATTR_GPS_ACCURACY)
+        last_seen = cast(datetime | None, attr.get(ATTR_LAST_SEEN))
+        self._prv_seen = last_seen
         self._attributes = {
-            ATTR_ALTITUDE: attr.get(ATTR_ALTITUDE),
             ATTR_ACTIVITY: attr.get(ATTR_ACTIVITY),
+            ATTR_ALTITUDE: attr.get(ATTR_ALTITUDE),
+            ATTR_BATTERY_CHARGING: attr.get(ATTR_BATTERY_CHARGING),
```

Comment:
Battery charging should be a binary sensor entity.

Reply:
I don't care so much about this piece of data, but why make this special when all other miscellaneous data are attributes?

Reply:
It's our current standard to put all measurements that are relevant on their own as separate sensor entities.

Reply:
So, does that mean I need to pull out all the others, too, and make them separate sensor entities? At a minimum, shouldn't that also apply to activity, altitude, battery_level, direction & speed? And can I assume, since you did not reply to my comments about last_seen, it's ok to leave it as an attribute on the device_tracker entity?

Reply:
Existing attributes don't need to be moved at this time, since that would be a breaking change.

We don't allow new additions of attributes that should be sensors.

Last seen is not allowed as an attribute.

Reply:
Then there's no point.